### PR TITLE
Logging in FacadeService and ResultHandler

### PR DESCRIPTION
**Pull request by tsehelnyk**
Added logging to FacadeService and ResultHandler fixes in XS2A-protocol, HBCI-protocol logging

### CODE REVIEW

File: opba-banking-protocol-facade/pom.xml [hunk 1]
```diff
@@ -23,6 +23,11 @@
             <artifactId>lombok</artifactId>
         </dependency>
 
+        <dependency>
```
Comment (valb3r) [inline opba-banking-protocol-facade/pom.xml:26]: Unnecessary dependency

File: opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/services/FacadeService.java [hunk 2]
```diff
@@ -25,17 +26,20 @@
...
 
         ProtocolWithCtx<ACTION, REQUEST> protocolWithCtx = txTemplate.execute(callback -> {
             InternalContext<REQUEST, ACTION> contextWithoutProtocol = contextFor(request);
             InternalContext<REQUEST, ACTION> ctx = selectAndSetProtocolTo(contextWithoutProtocol);
             ServiceContext<REQUEST> serviceContext = addRequestScopedFor(request, ctx);
             return new ProtocolWithCtx<>(ctx.getAction(), serviceContext);
         });
-        //TODO-1201 request's protocol + action
+
+        logResolver.log("Result of TransactionTemplate execution: {}", protocolWithCtx);
```
Comment (valb3r) [inline opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/services/FacadeService.java:41]: `Result of TransactionTemplate execution` -> `Result of protocol execution`

File: opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/services/ProtocolResultHandler.java [hunk 3]
```diff
@@ -55,6 +56,7 @@
     private final TokenBasedAuthService authService;
     private final TppTokenProperties tppTokenProperties;
     private final List<? extends ResultBodyPostProcessor> postProcessors;
+    private final FacadeLogResolver logResolver = new FacadeLogResolver(getClass());
```
Comment (valb3r) [inline opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/services/ProtocolResultHandler.java:59]: Can be static

File: opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/util/logresolver/domain/ProtocolWithCtxLog.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+                + ", context=" + (null != context ? context.getNotSensitiveData() : NULL)
+                + ")";
+    }
+
+    @SneakyThrows
+    @Override
+    public String toString() {
+
+        return "ServiceContextLog{"
+                + "protocol=" + protocol.toString()
```
Comment (valb3r) [inline opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/util/logresolver/domain/ProtocolWithCtxLog.java:32]: Null safety?

File: opba-banking-protocol-facade/pom.xml (hunk 1 above)
Comment (tsehelnyk) [reply opba-banking-protocol-facade/pom.xml:26]: dependency removed

File: opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/util/logresolver/domain/ProtocolWithCtxLog.java (hunk 4 above)
Comment (tsehelnyk) [reply opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/util/logresolver/domain/ProtocolWithCtxLog.java:32]: null check added

File: opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/util/logresolver/domain/request/RequestLog.java [hunk 5]
```diff
@@ -33,7 +33,7 @@ public String toString() {
 
         return "RequestLog{"
                 + "requestClass=" + request.getClass()
```
[the remark this answers is not in the document]
Comment (tsehelnyk) [reply opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/util/logresolver/domain/request/RequestLog.java:35]: null check added

File: opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/util/logresolver/domain/request/TransactionRequestLog.java [hunk 6]
```diff
@@ -13,7 +13,7 @@
     public String getNotSensitiveData() {
         return "TransactionRequestLog("
                 + "bankApi=" + request.getBankApiUser().getBankApi()
```
[the remark this answers is not in the document]
Comment (tsehelnyk) [reply opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/util/logresolver/domain/request/TransactionRequestLog.java:15]: refactoring, null check added

File: opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/util/logresolver/Constants.java [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+
+/**
+ * LogResolver constant names.
+ */
+
+
+@UtilityClass
+@SuppressWarnings("checkstyle:HideUtilityClassConstructor") // Lombok generates private ctor.
+public class Constants {
```
Comment (valb3r) [inline opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/util/logresolver/Constants.java:13]: Move `Constants` to `opba-facade-protocol-api-shared`

**Review by valb3r — APPROVED** [review]
Looks OK, but please fix minor comment

File: opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/util/logresolver/Constants.java (hunk 7 above)
Comment (tsehelnyk) [reply opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/util/logresolver/Constants.java:13]: done

**Pull request merged**